device/samples/javascript/readme.md

```diff
@@ -0,0 +1,174 @@
+
+**2) Grab the right files**
+
+Once you have the prerequisites, get the following files from the current folder:
+* **package.json**
+* **__sample_file.js__** (where **__sample_file.js__** is one of the files listed above and available in this folder)
+
+Copy them into your directory of choice. Once they've been placed in that directory, navigate using a terminal window to the folder. Then execute the following command:
+```
+npm install
+```
+
+This should install the proper dependencies as specified in the **package.json** you copied.
+
+**3) Set the DEVICE_CONNECTION_STRING environment variable**
+
+* From a shell or Node.js command prompt, navigate to the folder where you placed the sample files. 
+* Set the `DEVICE_CONNECTION_STRING` environment variable: 
+
+*in bash*
+```bash
+export DEVICE_CONNECTION_STRING="<YourIoTHubConnectionString>"
+```
+*in powershell*
+```powershell
+$env:DEVICE_CONNECTION_STRING="<YourIoTHubConnectionString>"
+```
+
+
+**4) Run it**
+
+Run the sample application using the following commands:
+
+```bash
+npm install
+node sample_file.js
+```
+
+## Read this if you want to run a sample in VS Code
+
+**1) Do steps 1 and 2 above**
+
+**2) Open Visual Studio Code**
+
+Open the folder containing your Node.js project in Visual Studio Code.
+
+**3) Set up the Debug Configuration**
+
+* Click on the *Debug* tab. (also `ctrl-shift-D`)
+* Find the little gear and click on it (When hovered over it says "Open launch.json")
+* Add the following launch configuration:
+
+```json
+{
+   "type": "node",
+   "request": "launch",
+   "name": "IoT Hub Sample Device",
+   "program": "${workspaceFolder}/device/samples/simple_sample_device.js",
+   "protocol": "inspector",
+   "sourceMaps": true,
+   "console": "integratedTerminal",
+   "env": {
+      "DEBUG": "mqtt*,rhea*,azure*",
+      "NODE_ENV": "development",
+      "DEVICE_CONNECTION_STRING": "<YourConnectionString>"
+   },
+   "outFiles": [
+      "${workspaceFolder}/common/core/lib/*.js",
+      "${workspaceFolder}/common/transport/amqp/lib/*.js",
+      "${workspaceFolder}/common/transport/http/lib/*.js",
+      "${workspaceFolder}/common/transport/mqtt/lib/*.js",
+      "${workspaceFolder}/device/core/lib/*.js",
+      "${workspaceFolder}/device/transport/amqp/lib/*.js",
+      "${workspaceFolder}/device/transport/http/lib/*.js",
+      "${workspaceFolder}/device/transport/mqtt/lib/*.js",
+      "${workspaceFolder}/dtclient/lib/*.js",
+      "${workspaceFolder}/service/lib/*.js",
+   ]
+}
+```
+* Edit the `DEVICE_CONNECTION_STRING` environment variable to use your IoT Hub device connection string.
+* Edit the `program` to the path to the code you want to run. 
+* If you do not want debug logs to be printed, delete the `DEBUG` environment variable. Otherwise you can leave it as it is.
+
+**4) Run it**
+
+* In the debug tab, click on the configuration drop down and select `IoT Hub Sample Device`, then press `Start Debugging`.
+
+
+## List of samples
```

Comment:
we might need to update this list.....

Reply:
List looks good to me. I updated the blob storage items. Fixed the extension in the TS examples.